packages/elements-core/src/components/TryIt/Body/FormDataBody.tsx

```diff
@@ -24,61 +25,120 @@ export const FormDataBody: React.FC<FormDataBodyProps> = ({
                 key={parameter.name}
                 parameter={parameter}
-                value={value instanceof File ? value : undefined}
-                onChange={newValue =>
-                  newValue
-                    ? onChangeValues({ ...values, [parameter.name]: newValue })
-                    : onChangeValues(omit(values, parameter.name))
+                value={typeof value === 'string' ? value : undefined}
+                onChange={value =>
+                  onChangeValues({
+                    ...values,
+                    [parameter.name]: typeof value === 'number' ? String(value) : (value as any),
+                  })
                 }
+                onChangeOptional={value => onChangeParameterAllow({ ...isAllowedEmptyValues, [parameter.name]: value })}
+                canChangeOptional={true}
+                isOptional={isAllowedEmptyValues[parameter.name] ?? false}
               />
             );
-          }
-
-          return (
-            <ParameterEditor
-              key={parameter.name}
-              parameter={parameter}
-              value={typeof value === 'string' ? value : undefined}
-              onChange={value =>
-                onChangeValues({
-                  ...values,
-                  [parameter.name]: typeof value === 'number' ? String(value) : (value as any),
-                })
-              }
-              onChangeOptional={value => onChangeParameterAllow({ ...isAllowedEmptyValues, [parameter.name]: value })}
-              canChangeOptional={true}
-              isOptional={isAllowedEmptyValues[parameter.name] ?? false}
-            />
-          );
-        })}
+          })}
       </Panel.Content>
     </Panel>
   );
 };
+
+export interface OneOfMenuProps {
+  /** all of the available sub-schemas that can be picked */
+  choices: Choice[];
+
+  /** the currently selected sub-schema */
+  choice: Choice;
+
+  /** called when the selection changes */
+  onChange: (choice: Choice) => void;
+}
+
+/**
+ * When the top level schema is `oneOf`, a drop-down menu that allows the user
+ * to select among the sub-schemas; otherwise `null`.
+ */
+export function OneOfMenu({ choices: subSchemas, choice: Choice, onChange }: OneOfMenuProps) {
+  const onSubSchemaSelect = React.useCallback(onChange, [onChange]);
+
+  const menuItems = React.useMemo(
+    () =>
+      subSchemas.map((subSchema, index) => {
+        const label = subSchema.title;
+        return {
+          id: `request-subschema-${label}`,
+          title: label,
+          onPress: () => onSubSchemaSelect(subSchema),
+        };
+      }),
+    [subSchemas, onSubSchemaSelect],
+  );
+
+  if (!subSchemas || subSchemas.length < 2) {
+    return null;
+  }
+
+  return (
+    <Menu
+      aria-label="Variants"
+      items={menuItems}
+      renderTrigger={({ isOpen }) => (
+        <Button appearance="minimal" size="sm" iconRight={['fas', 'sort']} active={isOpen} data-testid="oneof-menu">
+          Variants
```

Comment:
This is something to consider. We could display the selected name here (if it exists). We do that for Auth schemes

Reply:
Thanks, that's a good idea.  I made that change in 0954f90b.